### PR TITLE
Added Painting Fence Algorithm in Python

### LINKED ISSUE #3288: Painting Fence Algorithm in Python
## 💥 Proposal

I would like to implement Painting Fence Algorithm in Python as a part of Gssoc21.  

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

### PR DESCRIPTION
**Pull request by nandakishormpai**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
Yes

### Description

Added the code for Painting Fence Algorithm in Python

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes #3288

### CODE REVIEW

**Comment by nandakishormpai** [comment]
@HarshCasper @bhav09 @ankitaggarwal23  Kindly review my pull request.

File: Python/cp/Paint_Fence.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+"""
+___Paint Fence Algorithm___
+
+Problem:
+Given a fence with n posts and k colors
+Find out the number of ways of painting the fence
+such that at most 2 adjacent posts have the same color.
+
+Time Complexity  O(n)
```
Comment (ankitaggarwal23) [inline Python/cp/Paint_Fence.py]: Shift Time Complexity to the end and also add Space complexity

File: Python/cp/Paint_Fence.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+# 10
+# Enter k,Number of Colors
+# 4
+# Painting is not possible following the condition
+
+# Enter n,Number of Posts
+# 5
+# Enter k,Number of Colors
+# 3
+# Number of Possiblities: 180
```
Comment (ankitaggarwal23) [inline Python/cp/Paint_Fence.py:68]: Add the new line at the EOF

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: Python/cp/Paint_Fence.py (hunk 2 above)
Comment (nandakishormpai) [inline Python/cp/Paint_Fence.py:64]: Thank you for the feedback, i will make required  changes asap.

**Comment by nandakishormpai** [comment]
@ankitaggarwal23 i have made necessary changes as you advised. can you kindly review them ?

File: Python/cp/Paint_Fence.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+# Time Complexity  O(n)
+# Space Complexity O(n)
```
Comment (ankitaggarwal23) [inline Python/cp/Paint_Fence.py:67]: - Add a new line after SC.

File: Python/cp/Paint_Fence.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+___Paint Fence Algorithm___
+
+Problem:
+Given a fence with n posts and k colors
+Find out the number of ways of painting the fence
+such that at most 2 adjacent posts have the same color.
+
+
+
+
```
Comment (ankitaggarwal23) [inline Python/cp/Paint_Fence.py]: Remove that extra 3 line spaces

File: Python/cp/Paint_Fence.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+Input:
+n : number of posts
+k : number of colors
+
+Output:
+result: number of ways of painting
+        considering the condition above
+
+
```
Comment (ankitaggarwal23) [inline Python/cp/Paint_Fence.py]: Remove that extra spaces

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Do these changes and rest looks good to merge.

**Comment by nandakishormpai** [comment]
@ankitaggarwal23 ,I have made necessary changes.But when i add a new line at the end i am getting failed by the bot.

**Comment by nandakishormpai** [comment]
It is giving Multiple blank lines at the end of file

**Comment by nandakishormpai** [comment]
Is this fine? or should i make any further changes?

**Review by ankitaggarwal23 — APPROVED** [review]

**Comment by ankitaggarwal23** [comment]
> Is this fine? or should i make any further changes?

Fine, Leave it Looks good. You can ignore it, there is some glitch at deepsource's end.

**Comment by nandakishormpai** [comment]
Thank you

**Comment by nandakishormpai** [comment]
Sorry to bother you, I dont see my PR getting a score in my profile. Should i do anything more?

**Comment by ankitaggarwal23** [comment]
> Sorry to bother you, I don't see my PR getting a score in my profile. Should i do anything more?

@nandakishormpai2001 Actually scoring is not adding now and labelling will be added to each PR later. Don't worry about that.

**Pull request merged**